File: src/main/java/com/arangodb/intellij/aql/grammar/custom/psi/AqlNamedElement.java
package com.arangodb.intellij.aql.grammar.custom.psi;

import com.intellij.psi.NavigatablePsiElement;
import com.intellij.psi.PsiNameIdentifierOwner;
import com.intellij.psi.PsiNamedElement;
import org.jetbrains.annotations.NotNull;

public interface AqlNamedElement extends PsiNameIdentifierOwner, NavigatablePsiElement, PsiNamedElement {

    @NotNull
    AqlMixinType getAqlType();
}

File: src/main/kotlin/com/arangodb/intellij/aql/grammar/custom/psi/AqlNamedElement.kt
package com.arangodb.intellij.aql.grammar.custom.psi

import com.intellij.psi.NavigatablePsiElement
import com.intellij.psi.PsiNameIdentifierOwner
import com.intellij.psi.PsiNamedElement

interface AqlNamedElement : PsiNameIdentifierOwner, NavigatablePsiElement, PsiNamedElement {
    val aqlType: AqlMixinType
}
